staging_vespalib/src/vespa/vespalib/util/sequencedtaskexecutor.cpp

```diff
@@ -103,34 +105,59 @@ ExecutorStats
 SequencedTaskExecutor::getStats()
 {
     ExecutorStats accumulatedStats;
-    for (auto &executor :* _executors) {
+    for (auto &executor : _executors) {
         accumulatedStats.aggregate(executor->getStats());
     }
     return accumulatedStats;
 }
 
 ISequencedTaskExecutor::ExecutorId
 SequencedTaskExecutor::getExecutorId(uint64_t componentId) const {
-    uint32_t shrunkId = componentId % _component2Id.size();
-    uint8_t executorId = _component2Id[shrunkId];
+    PerfectKeyT key = componentId;
+    auto found = std::find(_component2IdPerfect.begin(), _component2IdPerfect.end(), key);
+    if (found != _component2IdPerfect.end()) {
+        return ExecutorId((found - _component2IdPerfect.begin()) % getNumExecutors());
+    } else if (_component2IdPerfect.size() < _component2IdPerfect.capacity()) {
+        return getExecutorIdPerfect(componentId);
+    } else {
+        return getExecutorIdImPerfect(componentId);
+    }
+}
+
+ISequencedTaskExecutor::ExecutorId
+SequencedTaskExecutor::getExecutorIdPerfect(uint64_t componentId) const {
+    PerfectKeyT key = componentId;
+    std::lock_guard guard(_mutex);
+    auto found = std::find(_component2IdPerfect.begin(), _component2IdPerfect.end(), key);
+    if (found == _component2IdPerfect.end()) {
+        _component2IdPerfect.push_back(key);
```

Comment:
Might need some protection against pushing too many elements onto `_component2IdPerfect`. If the old size is the same as `getNumExecutors() * 8 - 1` and to threads tries to get executor id for two new unique component ids, then
both might extend the vector, causing an unsafe resize (considering that it's read without lock).

Another issue is that `std::find()` in `getExecutorId()` might access uninitialized memory if `end()` returns new end before new value is written. If that uninitialized memory matches the key used for the search then we get an invalid result.